feat(nav): toggle dropdown arrow direction on open

sarthakNITT opened the pull request:
**Description**
fixed #4299 

Changes -
Add isOpen prop to NavItem to track dropdown state
Swap between down-arrow and up-arrow SVG in both <Link> and <button> branches
Mirror the existing down-arrow path on the X-axis for the up-arrow icon
Pass isOpen={open === '<menu>'} for Docs, Tools, and Community items in NavBar
Ensure arrow flips immediately when menu opens or closes on hover/click

Files changed -
Created ./components/icons/NavItemDropdownUp.tsx
Changed ./components/navigation/NavItem.tsx
Changed ./components/navigation/NavBar.tsx

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Navigation menus now show an “up” dropdown indicator when a section is expanded (Docs/Learning, Tools/Tooling, Community, Others, Language) and main nav items reflect open state.

- **Accessibility**
  - Menu items now expose expanded/collapsed state to assistive technologies for improved screen-reader support.

- **Style**
  - Updated dropdown icon visuals for clearer feedback when menus are open.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Sarthak114123082 committed:
Fixed lint erros

sambhavgupta0705 commented on components/navigation/NavBar.tsx:
```diff
@@ -60,11 +60,7 @@ export default function NavBar({ className = '', hideLogo = false }: NavBarProps
-    text: lang,
-    value: lang
-  }));
+  const uniqueLangs = getUniqueLangs().map((lang) => ({ key: lang, text: lang, value: lang }));
```
can we revert this formatting

sambhavgupta0705 commented on components/navigation/NavBar.tsx:
```diff
@@ -48,7 +48,7 @@ export default function NavBar({ className = '', hideLogo = false }: NavBarProps
     let pathnameWithoutLocale = pathname;
 
-    // Check if the pathname includes "/[lang]", if so, replace it with an empty string
+    // Check if the pathname includes '/[lang]', if so, replace it with an empty string
```
this one also

sambhavgupta0705 commented:
@sarthakNITT this doesn't work for mobile view

TRohit20 committed:
Merge branch 'master' into issue#4299

Sarthak114123082 committed:
Fix: applied changes as per instructions

Sarthak114123082 committed:
fix: fixed lint errors

sarthakNITT commented:
@sambhavgupta0705, Made changes. Please have a review and let me know if any further changes are needed.

Changes -
Fixed all the errors as intruceted.
Replaced <NavItemDropdownUp />  with {open === 'learning' ? <NavItemDropdownUp /> : <NavItemDropdown />} for all 5 navItems. Now working in mobile view.
Reorder imports in  ./components/navigation/MobileNavMenu.tsx to fix lint errors

Files changed -
Changed ./components/navigation/MobileNavMenu.tsx
Changed ./components/navigation/NavItem.tsx
Changed ./components/navigation/NavBar.tsx

Also on testing lint following warning were shown from the files I didn't touched.
<img width="965" height="490" alt="Screenshot 2025-08-11 at 2 42 17 PM" src="https://github.com/user-attachments/assets/612c1679-897c-4d38-b9a7-d3c9070696b3" />

Sarthak114123082 committed:
fix: added aria-expanded={isOpen} in NavItem

sarthakNITT commented:
@sambhavgupta070, Sorry for multiple commits. 

Verified all tests pass and lint error are fixed.
<img width="258" height="75" alt="Screenshot 2025-08-11 at 2 56 40 PM" src="https://github.com/user-attachments/assets/ed0e6603-8f20-4c57-9118-8cfa0e401e46" />

Sarthak114123082 committed:
fix: fixed lint error

Sarthak114123082 committed:
fix: added aria-haspopup in NavItem.tsx

Sarthak114123082 committed:
fix: fixed lint errors
